### ISSUE TITLE
A general question about training process

### ISSUE THREAD

**Issue by Saleh-Gholam-Zadeh** [polimi-ispl/icpr2020dfdc#13]
I have a some real images, which are not from any videos  and  I want to use them in the training phase, however they are not associated with any deepfake frame or video. I faced several questions

1) Is it necessary for any fake frame that the original frame is available in the dataset? How about the reverse question. Is it necessary for any original frame in the dataset that the fake frame is available in the dataset

2) How can I train a model when instead of videos I have some images, specifically I want to know how to create the face_dataframe for the images I have. Those images in my dataset are not associated with any videos.  

Best regards,
Saleh

**Comment by CrohnEngineer** [comment]
Hey @Saleh-Gholam-Zadeh ,

> Is it necessary for any fake frame that the original frame is available in the dataset? How about the reverse question. Is it necessary for any original frame in the dataset that the fake frame is available in the dataset

That is not necessary.

> How can I train a model when instead of videos I have some images, specifically I want to know how to create the face_dataframe for the images I have. Those images in my dataset are not associated with any videos.

There are several solutions already avaible among the [Torchvision datasets](https://pytorch.org/docs/stable/torchvision/datasets.html) API.
I would suggest you to use the [ImageFolder](https://pytorch.org/docs/stable/torchvision/datasets.html#imagefolder) dataset, where you just need to place all your images into two different folders based on labels inside the same directory (i.e. `dataset/FAKE` and `dataset/REAL`). The API manages the rest :)
Hope it helps!

Edoardo

**Comment by Saleh-Gholam-Zadeh** [comment]
Maybe I'm wrong but I was thinking it would be easier if I create the face_dataframe and don't change the train_binclass.py
However I want to make sure which columns of it are mandatory and which ones are not. for example as I said sometime the original frame associated with a fake frame can be unknown. 
lets say I have some fake images and some real images in 2 separate folders and I want to create the face_dataframe.
I would be thankful if you would help me with that. 
If you think "Imagefolder" solution is better can you please tell me roughly which part of the code should be modifed in train_binclass.py

Thanks a lot for your support and help

**Comment by CrohnEngineer** [comment]
> Maybe I'm wrong but I was thinking it would be easier if I create the face_dataframe and don't change the train_binclass.py

The FrameFaceDataset has been developed to handle specifically the data loading pipeline of the paper. 
For your scenario, I think it would be _much_ easier to work with the ImageFolder dataset already available in TorchVision than to modify our Dataset, also because, beside the data loading procedure, all the code for training the network in `train_binclass.py` is pretty standard.

> If you think "Imagefolder" solution is better can you please tell me roughly which part of the code should be modifed in train_binclass.py

The only part of the `train_binclass.py` to modify is the instantiation of the Dataset and of the DataLoader, so the code from line 212 to line 251.
This [post](https://towardsdatascience.com/pytorch-vision-multiclass-image-classification-531025193aa) explain how to use the ImageFolder and how to construct a training and validation set from just one directory of images like the one you have. 
It is for a multi-class classification, but the data loading should work also for a binary classification case like yours.
Assuming that your images are already cropped so that the face of the subject is centered, the other modification you need to do is to override the ImageFolder `__getitem__` method to work with Albumentations operations (natively, TorchVision datasets work with TorchVision transforms), something along this line:

```
from torchvision.datasets import ImageFolder

class AlbumentationsImageFolder(ImageFolder):
    """
    Simple extension to torchvision.ImageFolder to handle Albumentation transformations
    """

    def __getitem__(self, index: int) -> Tuple[Any, Any]:
        """
        Args:
            index (int): Index

        Returns:
            tuple: (sample, target) where target is class_index of the target class.
        """
        path, target = self.samples[index]
        sample = self.loader(path)
        if self.transform is not None:
            sample = self.transform(image=sample)['image']  # modification following Albumentations transforms API
        if self.target_transform is not None:
            target = self.target_transform(target)

        return sample, target
```

You might also need to slightly modify the constructor method, but I'm not sure: for the rest, I think you have all the pointers to make your code run. 
I really think this solution is a lot faster than modifying our code, you just need to get your hands a little dirty :)
Hope it helps!

Edoardo

**Comment by Saleh-Gholam-Zadeh** [comment]
Thanks a lot for your help, 
of course your post will help.
Before you answer me, I tried something else, I used ffmpeg to create a video using images I have, and set frame reate to 1 Frame per second, then split the whole video to some vidoes with the length of  32 seconsd (containing 32 images), then I followed the normal procedure, Im wondering if it makes sense or not, or Im missing something?
